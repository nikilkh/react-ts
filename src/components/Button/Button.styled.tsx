import styled from "styled-components";

export const ButtonStyle = styled.button`
  all: unset;
  background-color: #352c9a;
  font-family: "Rubik", sans-serif;
  font-size: 16px;
  font-weight: 600;
  color: white;
  padding: 12px 34px;
  cursor: pointer;

  &:active {
    text-decoration: underline;
  }

  &:disabled {
    background: #626262;
  }

  &:hover {
    background: #724fd8;
  }
`;
