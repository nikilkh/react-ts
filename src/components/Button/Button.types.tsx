export interface buttonProps {
    children: React.ReactNode;
    onclick: React.MouseEventHandler<HTMLButtonElement>;
  }