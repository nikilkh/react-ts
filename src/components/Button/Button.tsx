import { ButtonStyle } from "./Button.styled";
import { buttonProps } from "./Button.types";

export const Button = (props: buttonProps) => {
  return <ButtonStyle onClick={props.onclick}>{props.children}</ButtonStyle>;
};
