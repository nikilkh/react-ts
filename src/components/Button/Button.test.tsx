import { render } from "@testing-library/react";
import { execPath } from "process";
import { Button } from "./Button";

const mockClickHandle = jest.fn();

describe("renders learn react link", () => {
  it("should render hello world", () => {
    const { container } = render(
      <Button onclick={mockClickHandle} children="click me" />
    );
    expect(container).toMatchSnapshot();
    
  });
});
