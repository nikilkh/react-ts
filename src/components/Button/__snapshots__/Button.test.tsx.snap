// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link should render hello world 1`] = `
<div>
  <button
    class="sc-bczRLJ cgVelk"
  >
    click me
  </button>
</div>
`;
